java-checks/src/main/java/org/sonar/java/filters/BaseTreeVisitorIssueFilter.java

```diff
@@ -83,7 +83,8 @@ public boolean accept(Issue issue) {
     return true;
   }
 
-  public void ignoreIssuesInTree(Tree tree, Class<? extends JavaCheck> filteredRule) {
+  @Override
+  public void markIssues(Tree tree, Class<? extends JavaCheck> filteredRule, IssueState state) {
```

Comment:
Name is misleading as this method as nothing to do with issues : I would go for something like : computeFilteredLinesForRule or something alike. And maybe have two methods: one for inclusion, and one for exclusion with an internal flag to simplify the API from caller perspective.